src/main/resources/org/jpeek/metrics/MWE.xsl

```diff
@@ -0,0 +1,104 @@
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-->
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="2.0">
+  <xsl:template match="skeleton">
+    <metric>
+      <xsl:apply-templates select="@*"/>
+      <title>MWE</title>
+      <description>
+        <xsl:text><![CDATA[
+          MWE = max( Oi * Di )
+
+          where:
+          Oi = sum( Pdi ) / n: Occupancy of topic i in a class;
+          Di = sum( -Qdi * log(Qdi) ) / log(n): Distribution of topic i in a class;
+          Qdi = Pdi / sum( Pdi ): distribution of topic i in a method d;
+          Pdi - probability of topic i in a method d;
+          n - number of methods.
+        ]]></xsl:text>
+      </description>
+      <xsl:apply-templates select="node()"/>
+    </metric>
+  </xsl:template>
+  <xsl:template match="class">
+    <!--
+    @todo #17:30min The current implementation of MWE metric expects information
+     about LDA topics and their probability in skeleton.xml by xpath `class/methods/method/topics/topic`.
+     `topic` items must have two mandatory attributes - `@name` (topic unique identifier) and `@p` (probability of the
+     topic's occurence in the method). Due to #129, this part of information is missing.
+     Ensure that the core part of JPeek provides the information about topics and their probability
+     and update the metric in accordance with the provided XML structure.
+    -->
+    <xsl:variable name="n" select="count(methods/method)"/>
+    <xsl:variable name="Oi">
+      <xsl:for-each-group select="methods/method/topics/topic" group-by="@name">
+        <xsl:variable name="topic" select="current-grouping-key()"/>
+        <oi topic="{$topic}" value="{sum(current-group()/@p) div $n}"/>
+      </xsl:for-each-group>
+    </xsl:variable>
+    <xsl:variable name="Qdi">
+      <xsl:variable name="cur" select="."/>
+      <xsl:for-each-group select="methods/method" group-by="@name">
+        <xsl:variable name="method" select="current-grouping-key()"/>
+        <xsl:for-each-group select="current-group()/topics/topic" group-by="@name">
+          <xsl:variable name="topic" select="current-grouping-key()"/>
+          <xsl:variable name="pdi" select="$cur/methods/method[@name=$method]/topics/topic[@name=$topic]/@p"/>
+          <xsl:variable name="pdSum" select="sum($cur/methods/method/topics/topic[@name=$topic]/@p)"/>
+          <qdi method="{$method}" topic="{$topic}" value="{$pdi div $pdSum}"/>
+        </xsl:for-each-group>
+      </xsl:for-each-group>
+    </xsl:variable>
+    <xsl:variable name="Di">
+      <!--
+      @todo #17:30min Di value is supposed to be calculated by formula:
+       Di = sum( -Qdi * log(Qdi) ) / log(n).
+       See the MWE metric description for details.
+       However, logarithm function is currently not available in XSL (issue #128), that's why it is stubbed.
```

Comment:
@skapral no need to create an issue, 0pdd will do it automatically

Reply:
@yegor256 Issue was on adding logarithm capability in general, while puzzle is on unstubbing `Di`. But ok, since the issue is closed, I'll update the puzzle and remove the reference from it.

Reply:
Updated